File: Common/Models/Payment/PaymentInfo.cs
using System.ComponentModel.DataAnnotations;

namespace Common.Models.Payment
{
    public record PaymentInfo
    {
        [Required]
        public required string CardName { get; set; }

        [Required]
        public required string CardNumber { get; set; }

        [Required]
        public required string Expiration { get; set; }

        [Required]
        public required string CVV { get; set; }

        [Required]
        public required PaymentMethod PaymentMethod { get; set; }
    }
}

File: rorderapi/src/models/paymentinfo.rs
use bson::doc;
use serde::{Serialize, Deserialize};

use super::paymentmethod::PaymentMethod;

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "PascalCase")]
pub struct PaymentInfo {
    pub card_name: String,
    pub card_number: String,
    pub expiration: String,
    #[serde(rename = "CVV")] 
    pub cvv: String,
    pub payment_method: PaymentMethod
}